libparsec/crates/platform_http_proxy/src/native/windows.rs

```diff
@@ -0,0 +1,80 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use widestring::U16CStr;
+use windows_sys::Win32::{
+    Foundation::GetLastError,
+    Networking::WinHttp::{
+        WinHttpGetIEProxyConfigForCurrentUser, WINHTTP_CURRENT_USER_IE_PROXY_CONFIG,
+    },
+    System::Memory::GlobalFree,
+};
+
+use libparsec_types::anyhow;
+
+use crate::ProxyConfig;
+
+impl ProxyConfig {
+    pub(crate) fn with_register(self) -> anyhow::Result<Self> {
+        let mut cfg = self;
+
+        let mut proxy_config = WINHTTP_CURRENT_USER_IE_PROXY_CONFIG {
+            fAutoDetect: 0,
+            lpszAutoConfigUrl: std::ptr::null_mut(),
+            lpszProxy: std::ptr::null_mut(),
+            lpszProxyBypass: std::ptr::null_mut(),
+        };
+
+        // Safety: Nothing unsafe here
```

Comment:
Improve the comment, why is it safe here ? (hint we initialize the struct `proxy_config` at the previous step with sensible value)

Reply:
which sensible value ?

Reply:
`proxy_config` being initialize with null ptr for pointer data instead of a random value.

Reply:
We don't care about the parameters, because this is the output

Reply:
The only thing we should ensure is that the `pointer` provided is a `WINHTTP_CURRENT_USER_IE_PROXY_CONFIG` for the correct `size` but I don't cast anything (no other `unsafe block`) and `rust` provides the check on `type`

Reply:
The last comment would make a nice safety comment ;)

Reply:
Done